src/test/java/com/googlecode/protobuf/format/JsonFormatTest.java

```diff
@@ -52,8 +52,7 @@ public void testJsonFormat(
         assertThat(formatter.printToString(input), is(expectedString));
     }
 
-    // TODO(scr): Re-enable test when the code is fixed to enable slurping unknown fields into the message.
-    @Test(enabled = false, description = "https://github.com/bivas/protobuf-java-format/issues/23")
+    @Test(enabled = true, description = "https://github.com/bivas/protobuf-java-format/issues/23")
     public void testIssue23() throws Exception {
```

Comment:
Add another use case with more complex unknown things - i.e. an unknown message (if that's possible).

I think this may be what you need:

https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/UnknownFieldSet.Field.Builder#addGroup(com.google.protobuf.UnknownFieldSet)